From 7b4c3272f3a1532f61fbc05c253cc3f404bc734e Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Wed, 4 Dec 2024 02:48:07 +0800
Subject: [PATCH 129/132] AOSCOS: MIPS: Loongson: Fix constant timer requesting

In
"AOSCOS: MIPS: loongson64: fix constant timer build on kernel versions >= v5.7-rc2",
usage of setup_irq() was changed to request_irq(), however the dev_id in
latter was set to NULL, causing it to return -EINVAL combined with
IRQF_SHARED set in irqflags. Fix by passing the correct dev_id.

Fixes: "AOSCOS: MIPS: Loongson: Add constant timer support"
Fixes: "AOSCOS: MIPS: loongson64: fix constant timer build on kernel versions >= v5.7-rc2"
Cc: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/constant_timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/loongson64/constant_timer.c b/arch/mips/loongson64/constant_timer.c
index ca8b2c14602f..c1fda4a3ae16 100644
--- a/arch/mips/loongson64/constant_timer.c
+++ b/arch/mips/loongson64/constant_timer.c
@@ -190,7 +190,7 @@ int constant_clockevent_init(void)
 
 	err = request_irq(irq, constant_timer_interrupt,
 		    IRQF_PERCPU | IRQF_TIMER | IRQF_SHARED,
-		    "timer", NULL);
+		    "timer", cd);
 	if (err) {
 		pr_err("loongson64: setup irq for constant_clock_event failed: %d\n", err);
 		return err;
-- 
2.47.1

